IQSS/dataverse-frontend
# 316 - Functional create dataset form

Here's a revised version of the text with corrected English:

## What This PR Does / Why We Need It:
This PR introduces a functional 'Create Dataset' form, allowing users to create new datasets via the form. However, this version of the form has several limitations:

1. It only includes the required fields.
2. The metadata fields are not dynamically rendered; the UI is not connected to the backend for the configuration of these fields (TBD in #336 )
3. In the JSF version, required fields could be multiple (e.g., there could be several author names). However, in this first iteration of the SPA, users can only add one author name. (TBD in #337)
4. The validation is hardcoded, meaning that the SPA is not connected to the backend for metadata field validation. (TBD in #338)

This constitutes the first iteration of the 'Create Dataset' form. We plan to address these limitations in future updates.

## Which Issue(s) This PR Closes:
- Closes #316

## Special Notes for Your Reviewer:
I made a few changes to the Design System form fields.

Below are the issues created to continue with the development of this feature:
- #336 
- #337 
- #338 
- #339 
- #340 

## Suggestions on How to Test This:
### Step 1: Run the Development Environment
1. Execute `npm i`.
2. Navigate with `cd packages/design-system && npm run build`.
3. Return with `cd ../../`.
4. Ensure you have a `.env` file similar to `.env.example`, with the variable `VITE_DATAVERSE_BACKEND_URL=http://localhost:8000`.
5. Navigate with `cd dev-env`.
6. Start the environment using `./run-env.sh unstable`.
7. To verify the environment, visit <http://localhost:8000> and check your local Dataverse installation.

### Step 2: Test the Form
1. Navigate to <http://localhost:8000>.
2. Log in as an admin using the username: `dataverseAdmin` and password: `admin1`.
3. Go to <http://localhost:8000/spa> and click the 'Add Data >> Add Dataset' button.
4. Experiment with the form validation.
5. Complete the form and click 'Save Dataset'; you should be redirected to your new dataset.

## Does This PR Introduce a User Interface Change? If Mockups Are Available, Please Link/Include Them Here:
![image](https://github.com/IQSS/dataverse-frontend/assets/23359572/fc29df40-df65-4a42-abc0-aece08ba4a3e)

## Is There a Release Notes Update Needed for This Change?:
Added the 'Create dataset' feature to the SPA.

## Additional Documentation:
